interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,376 @@
+angular.module('adage.gene.search', [
+  'adage.gene.resource'
+])
+
+.config(function($stateProvider) {
+  $stateProvider
+    .state( 'gene_search', {
+      url: '/gene_search',
+      views: {
+        "main": {
+          templateUrl: 'gene/gene-network.tpl.html',
+          controller: ['$scope', 'UserFactory',
+            function($scope, UserFactory) {
+              UserFactory.getPromise().$promise.then( function() {
+                $scope.userObj = UserFactory.getUser();
+              });
```

Comment:
Right, the `.then()` is called when the promise is fulfilled, but not if it's rejected. You could cover the other case by including a second callback function in `.then()`, by using a `.catch()` (as described in that promise write-up I found: https://developers.google.com/web/fundamentals/getting-started/primers/promises?hl=en ), or you could simply initialize `$scope.userObj` to `null` explicitly before the call to `UserFactory.getPromise()`. I think the last solution is the simplest, as it's a one-line fix and easy to follow.